contracts/settling_game/modules/food/Food.cairo

```diff
@@ -0,0 +1,542 @@
+                Uint256(total_food * 10 ** 18, 0),
+                1,
+                data,
+            )
+        else:
+            # fish
+            IERC1155.mint(
+                resources_address,
+                owner,
+                Uint256(ResourceIds.fish, 0),
+                Uint256(total_food * 10 ** 18, 0),
+                1,
+                data,
+            )
+        end
+        tempvar syscall_ptr = syscall_ptr
+        tempvar range_check_ptr = range_check_ptr
+        tempvar pedersen_ptr = pedersen_ptr
+    else:
+        # turn directly into useable food
+        convert_to_store(token_id, total_food)
+        tempvar syscall_ptr = syscall_ptr
+        tempvar range_check_ptr = range_check_ptr
+        tempvar pedersen_ptr = pedersen_ptr
+    end
+
+    return ()
+end
+
+# -----------------------------------
+# INTERNAL
+# -----------------------------------
+
+# @notice Updates values after harvest
+# @param current_harvests: Staked Realm id (S_Realm)
+# @param number_built: farms built - this is just unpacked data from above
+# @param total_harvest: total qty harvested
+# @param decayed_farms: decayed farms
+# @param token_id: Staked Realm id (S_Realm)
+# @param food_building_id: food building id
+func update{
+    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr : BitwiseBuiltin*
+}(
+    current_harvests : felt,
+    number_built : felt,
+    total_harvest : felt,
+    decayed_farms : felt,
+    token_id : Uint256,
+    food_building_id : felt,
+):
+    alloc_locals
+    let (block_timestamp) = get_block_timestamp()
+
+    # check harvests left
+
+    with_attr error_message("FOOD: No Harvests available"):
+        assert_not_zero(current_harvests)
+    end
+
+    # calculated harvests to save
+    let harvests_to_save = current_harvests - total_harvest - decayed_farms
+
+    # # if 0 harvests left: Some can decay if you do not harvest.
+    let (is_zero_harvests) = is_le(harvests_to_save, 0)
+
+    if is_zero_harvests == TRUE:
+        tempvar harvests_to_save = 0
+    else:
+        tempvar harvests_to_save = harvests_to_save
+    end
+
+    let (packed) = Food.pack_food_buildings(
+        food_buildings_unpacked=FoodBuildings(number_built, harvests_to_save, block_timestamp)
+    )
+
+    # save packed
+    if food_building_id == RealmBuildingsIds.Farm:
+        farms.write(token_id, packed)
+    else:
+        fishing_villages.write(token_id, packed)
+    end
+
+    return ()
+end
+
+# @notice Converts harvest directly into food store
+# @param token_id: Staked Realm id (S_Realm)
+# @param quantity: quantity of food to store
+func convert_to_store{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256, quantity : felt
+):
+    alloc_locals
+    let (block_timestamp) = get_block_timestamp()
+
+    let (current) = food_in_store(token_id)
+
+    store_house.write(token_id, current + quantity + block_timestamp)
```

Comment:
Uhm, adding `block_timestamp` here doesn't look right 🧐

Reply:
this just updates the real food value to the computed value - this way when the population changes, we update the actual food supply to reflect it